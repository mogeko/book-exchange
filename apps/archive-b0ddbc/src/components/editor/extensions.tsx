import StarterKit from "@tiptap/starter-kit";
import CharacterCount from "@tiptap/extension-character-count";
import Highlight from "@tiptap/extension-highlight";
import Typography from "@tiptap/extension-typography";
import Placeholder from "@tiptap/extension-placeholder";
import FocusClasses from "@tiptap/extension-focus";
import Link from "@tiptap/extension-link";

const extensions = [
  StarterKit,
  Highlight.configure({
    multicolor: true,
  }),
  Typography,
  Placeholder.configure({
    emptyEditorClass: "is-editor-empty",
    placeholder: "Write something here...",
  }),
  Link.configure({
    protocols: ["ftp", "http", "https", "mailto"],
  }),
  FocusClasses,
  CharacterCount,
];

export default extensions;
