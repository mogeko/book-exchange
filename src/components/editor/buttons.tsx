import classNames from "@/lib/utils/classNames";
import { FaBold, FaItalic, FaStrikethrough, FaCircle } from "react-icons/fa";
import { FaListOl, FaListUl, FaCode, FaQuoteLeft } from "react-icons/fa";
import type { Editor } from "@tiptap/react";

export const Bold: React.FC<ButtonProps> = ({ editor }) => {
  return (
    <button
      onClick={() => editor.chain().focus().toggleBold().run()}
      className={classNames(
        editor.isActive("bold") ? "btn-active" : "",
        "btn btn-xs"
      )}
    >
      <FaBold className="w-3 h-3" />
    </button>
  );
};

export const Italic: React.FC<ButtonProps> = ({ editor }) => {
  return (
    <button
      onClick={() => editor.chain().focus().toggleItalic().run()}
      className={classNames(
        editor.isActive("italic") ? "btn-active" : "",
        "btn btn-xs"
      )}
    >
      <FaItalic className="w-3 h-3" />
    </button>
  );
};

export const Strike: React.FC<ButtonProps> = ({ editor }) => {
  return (
    <button
      onClick={() => editor.chain().focus().toggleStrike().run()}
      className={classNames(
        editor.isActive("strike") ? "btn-active" : "",
        "btn btn-xs"
      )}
    >
      <FaStrikethrough className="w-3 h-3" />
    </button>
  );
};

export const H1: React.FC<ButtonProps> = ({ editor }) => {
  return (
    <button
      onClick={() => editor.chain().focus().toggleHeading({ level: 1 }).run()}
      className={classNames(
        editor.isActive("heading", { level: 1 }) ? "btn-active" : "",
        "btn btn-xs"
      )}
    >
      H1
    </button>
  );
};

export const H2: React.FC<ButtonProps> = ({ editor }) => {
  return (
    <button
      onClick={() => editor.chain().focus().toggleHeading({ level: 2 }).run()}
      className={classNames(
        editor.isActive("heading", { level: 2 }) ? "btn-active" : "",
        "btn btn-xs"
      )}
    >
      H2
    </button>
  );
};

export const H3: React.FC<ButtonProps> = ({ editor }) => {
  return (
    <button
      onClick={() => editor.chain().focus().toggleHeading({ level: 3 }).run()}
      className={classNames(
        editor.isActive("heading", { level: 3 }) ? "btn-active" : "",
        "btn btn-xs"
      )}
    >
      H3
    </button>
  );
};

export const BulletList: React.FC<ButtonProps> = ({ editor }) => {
  return (
    <button
      onClick={() => editor.chain().focus().toggleBulletList().run()}
      className={classNames(
        editor.isActive("bulletList") ? "btn-active" : "",
        "btn btn-xs"
      )}
    >
      <FaListUl className="w-3 h-3" />
    </button>
  );
};

export const OrderedList: React.FC<ButtonProps> = ({ editor }) => {
  return (
    <button
      onClick={() => editor.chain().focus().toggleOrderedList().run()}
      className={classNames(
        editor.isActive("orderedList") ? "btn-active" : "",
        "btn btn-xs"
      )}
    >
      <FaListOl className="w-3 h-3" />
    </button>
  );
};

export const CodeBlock: React.FC<ButtonProps> = ({ editor }) => {
  return (
    <button
      onClick={() => editor.chain().focus().toggleCodeBlock().run()}
      className={classNames(
        editor.isActive("codeBlock") ? "btn-active" : "",
        "btn btn-xs"
      )}
    >
      <FaCode className="w-3 h-3" />
    </button>
  );
};

export const Blockquote: React.FC<ButtonProps> = ({ editor }) => {
  return (
    <button
      onClick={() => editor.chain().focus().toggleBlockquote().run()}
      className={classNames(
        editor.isActive("blockquote") ? "btn-active" : "",
        "btn btn-xs"
      )}
    >
      <FaQuoteLeft className="w-3 h-3" />
    </button>
  );
};

export const RedMark: React.FC<ButtonProps> = ({ editor }) => {
  return (
    <button
      onClick={() =>
        editor.chain().focus().toggleHighlight({ color: "#FFA8A8" }).run()
      }
      className={classNames(
        editor.isActive("highlight", { color: "#FFA8A8" }) ? "btn-active" : "",
        "btn btn-xs"
      )}
    >
      <FaCircle className="w-3 h-3 text-[#FFA8A8]" />
    </button>
  );
};

export const YellowMark: React.FC<ButtonProps> = ({ editor }) => {
  return (
    <button
      onClick={() =>
        editor.chain().focus().toggleHighlight({ color: "#FFC078" }).run()
      }
      className={classNames(
        editor.isActive("highlight", { color: "#FFC078" }) ? "btn-active" : "",
        "btn btn-xs"
      )}
    >
      <FaCircle className="w-3 h-3 text-[#FFC078]" />
    </button>
  );
};

export const GreenMark: React.FC<ButtonProps> = ({ editor }) => {
  return (
    <button
      onClick={() =>
        editor.chain().focus().toggleHighlight({ color: "#8CE99A" }).run()
      }
      className={classNames(
        editor.isActive("highlight", { color: "#8CE99A" }) ? "btn-active" : "",
        "btn btn-xs"
      )}
    >
      <FaCircle className="w-3 h-3 text-[#8CE99A]" />
    </button>
  );
};

export const BlueMark: React.FC<ButtonProps> = ({ editor }) => {
  return (
    <button
      onClick={() =>
        editor.chain().focus().toggleHighlight({ color: "#74C0FC" }).run()
      }
      className={classNames(
        editor.isActive("highlight", { color: "#74C0FC" }) ? "btn-active" : "",
        "btn btn-xs"
      )}
    >
      <FaCircle className="w-3 h-3 text-[#74C0FC]" />
    </button>
  );
};

export const PresetButtons = {
  bubble: [Bold, Italic, Strike, RedMark, YellowMark, GreenMark, BlueMark],
  floating: [H1, H2, H3, BulletList, OrderedList, CodeBlock, Blockquote],
};

export interface ButtonProps {
  editor: Editor;
}
