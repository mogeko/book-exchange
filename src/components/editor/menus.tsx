import classNames from "@/lib/utils/classNames";
import { BubbleMenu as Bubble, FloatingMenu as Floating } from "@tiptap/react";
import { FaLink } from "react-icons/fa";
import { useRef, useState, useEffect } from "react";
import { useCallback, useMemo } from "react";
import type { ButtonProps } from "@/components/editor/buttons";
import type { Editor } from "@tiptap/react";

export const BubbleMenu: React.FC<MenuProps> = ({ editor, buttons }) => {
  const [isLinkActive, setIsLinkActive] = useState(false);
  const [hasLinkInput, setHasLinkInput] = useState(false);
  const inputRef = useRef<HTMLInputElement>(null);
  const handleSubmit = useCallback(() => {
    if (inputRef.current?.value) {
      editor?.chain().focus().setLink({ href: inputRef.current?.value }).run();
    } else {
      editor?.chain().focus().unsetLink().run();
    }
    setIsLinkActive(!isLinkActive);
  }, [editor, isLinkActive, setIsLinkActive]);
  const LinkDialog = useMemo(() => {
    return (
      <div className="input-group">
        <input
          onChange={(e) => setHasLinkInput(e.target.value !== "")}
          defaultValue={editor?.getAttributes("link").href}
          className="input input-bordered input-xs"
          ref={inputRef}
          autoFocus
        />
        <button
          onClick={handleSubmit}
          className={classNames(
            hasLinkInput ? "btn-primary" : "",
            "btn btn-xs"
          )}
        >
          OK
        </button>
      </div>
    );
  }, [editor, hasLinkInput, handleSubmit]);
  const LinkButton = useMemo(() => {
    return (
      <button
        onClick={() => setIsLinkActive(!isLinkActive)}
        className={classNames(
          editor?.isActive("link") ? "btn-active" : "",
          "btn btn-xs"
        )}
      >
        <FaLink className="w-3 h-3" />
      </button>
    );
  }, [editor, isLinkActive, setIsLinkActive]);

  useEffect(() => {
    editor?.isFocused && setIsLinkActive(false);
  }, [editor?.isFocused, setIsLinkActive]);

  return (
    editor && (
      <Bubble
        className="btn-group"
        tippyOptions={{ duration: 100 }}
        editor={editor}
      >
        {!isLinkActive &&
          buttons.map((Button, i) => <Button key={i} editor={editor} />)}
        {!isLinkActive && LinkButton}
        {isLinkActive && LinkDialog}
      </Bubble>
    )
  );
};

export const FloatingMenu: React.FC<MenuProps> = ({ editor, buttons }) => {
  return (
    editor && (
      <Floating
        className="btn-group"
        tippyOptions={{ duration: 100 }}
        editor={editor}
      >
        {buttons.map((Button, i) => (
          <Button key={i} editor={editor} />
        ))}
      </Floating>
    )
  );
};

interface MenuProps {
  editor: Editor | null;
  buttons: React.FC<ButtonProps>[];
}
