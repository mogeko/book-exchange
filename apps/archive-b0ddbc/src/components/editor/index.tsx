import { BubbleMenu, FloatingMenu } from "@/components/editor/menus";
import { PresetButtons } from "@/components/editor/buttons";
import extensions from "@/components/editor/extensions";
import { useEditor, EditorContent } from "@tiptap/react";
import type { JSONContent } from "@tiptap/react";

const Editor: React.FC<EditorProps> = ({ onSubmit }) => {
  const editor = useEditor({
    extensions: extensions,
    editorProps: {
      attributes: {
        class:
          "textarea textarea-bordered textarea-accent h-full w-full overflow-auto prose",
        role: "textbox",
      },
    },
  });
  const handleClick = () => {
    onSubmit({ content: editor?.getJSON() });
    editor?.commands.clearContent();
  };

  return (
    <div className="relative max-w-xl">
      <EditorContent editor={editor} className="flex h-40 w-full" />
      <FloatingMenu editor={editor} buttons={PresetButtons.floating} />
      <BubbleMenu editor={editor} buttons={PresetButtons.bubble} />
      <div className="label items-start">
        <span className="label-text-alt">
          {editor?.storage.characterCount.characters()} characters
        </span>
        <button
          onClick={handleClick}
          className="btn btn-sm btn-primary"
          {...(editor?.isEmpty ? { disabled: true } : {})}
        >
          Submit
        </button>
      </div>
    </div>
  );
};

interface EditorProps {
  onSubmit: (data: EditorFormInput) => void;
}

export interface EditorFormInput {
  content?: JSONContent;
}

export default Editor;
