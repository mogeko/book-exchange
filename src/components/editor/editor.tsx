import { BubbleMenu, FloatingMenu } from "@/components/editor/menus";
import { PresetButtons } from "@/components/editor/buttons";
import { useEditor, EditorContent } from "@tiptap/react";
import StarterKit from "@tiptap/starter-kit";
import CharacterCount from "@tiptap/extension-character-count";
import Highlight from "@tiptap/extension-highlight";
import Typography from "@tiptap/extension-typography";
import Placeholder from "@tiptap/extension-placeholder";
import FocusClasses from "@tiptap/extension-focus";
import Link from "@tiptap/extension-link";

const Editor: React.FC = () => {
  const editor = useEditor({
    extensions: [
      StarterKit,
      Highlight.configure({
        multicolor: true,
      }),
      Typography,
      Placeholder.configure({
        emptyEditorClass: "is-editor-empty",
        placeholder: "Write something here...",
      }),
      Link.configure({
        protocols: ["ftp", "http", "https", "mailto"],
      }),
      FocusClasses,
      CharacterCount,
    ],
    editorProps: {
      attributes: {
        class:
          "textarea textarea-bordered textarea-accent h-full w-full overflow-auto prose",
      },
    },
  });
  const handleClick = () => {
    return;
  };

  return (
    <div className="relative max-w-xl">
      <EditorContent editor={editor} className="flex h-40 w-full" />
      <FloatingMenu editor={editor} buttons={PresetButtons.floating} />
      <BubbleMenu editor={editor} buttons={PresetButtons.bubble} />
      <div className="label items-start">
        <span className="label-text-alt">
          {editor?.storage.characterCount.characters()} characters
        </span>
        <button
          onClick={handleClick}
          className="btn btn-sm btn-primary"
          {...(editor?.isEmpty ? { disabled: true } : {})}
        >
          Submit
        </button>
      </div>
    </div>
  );
};

export default Editor;
