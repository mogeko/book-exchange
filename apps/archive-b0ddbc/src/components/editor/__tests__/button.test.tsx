import "@testing-library/jest-dom";
import { render, renderHook, screen, fireEvent } from "@testing-library/react";
import { PresetButtons } from "@/components/editor/buttons";
import extensions from "@/components/editor/extensions";
import { useEditor } from "@tiptap/react";

const {
  bubble: [Bold, Italic, RedMark, YellowMark, GreenMark, BlueMark, Strike],
  floating: [H1, H2, H3, BulletList, OrderedList, CodeBlock, Blockquote],
} = PresetButtons;

const { result } = renderHook(() => useEditor({ extensions }));

describe("Buttons", () => {
  it("render a Bold button", () => {
    const { container } = render(<Bold editor={result.current!} />);

    expect(screen.getByRole("button")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button"));

    expect(container).toMatchSnapshot();
  });

  it.skip("render a Italic button", () => {
    const { container } = render(<Italic editor={result.current!} />);

    expect(screen.getByRole("button")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button"));

    expect(container).toMatchSnapshot();
  });

  it.skip("render a RedMark button", () => {
    const { container } = render(<RedMark editor={result.current!} />);

    expect(screen.getByRole("button")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button"));

    expect(container).toMatchSnapshot();
  });

  it.skip("render a YellowMark button", () => {
    const { container } = render(<YellowMark editor={result.current!} />);

    expect(screen.getByRole("button")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button"));
    expect(container).toMatchSnapshot();
  });

  it.skip("render a GreenMark button", () => {
    const { container } = render(<GreenMark editor={result.current!} />);

    expect(screen.getByRole("button")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button"));

    expect(container).toMatchSnapshot();
  });

  it.skip("render a BlueMark button", () => {
    const { container } = render(<BlueMark editor={result.current!} />);

    expect(screen.getByRole("button")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button"));

    expect(container).toMatchSnapshot();
  });

  it.skip("render a Strike button", () => {
    const { container } = render(<Strike editor={result.current!} />);

    expect(screen.getByRole("button")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button"));

    expect(container).toMatchSnapshot();
  });

  it.skip("render a H1 button", () => {
    const { container } = render(<H1 editor={result.current!} />);

    expect(screen.getByRole("button")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button"));

    expect(container).toMatchSnapshot();
  });

  it.skip("render a H2 button", () => {
    const { container } = render(<H2 editor={result.current!} />);

    expect(screen.getByRole("button")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button"));

    expect(container).toMatchSnapshot();
  });

  it.skip("render a H3 button", () => {
    const { container } = render(<H3 editor={result.current!} />);

    expect(screen.getByRole("button")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button"));
    expect(container).toMatchSnapshot();
  });

  it.skip("render a BulletList button", () => {
    const { container } = render(<BulletList editor={result.current!} />);

    expect(screen.getByRole("button")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button"));

    expect(container).toMatchSnapshot();
  });

  it.skip("render a OrderedList button", () => {
    const { container } = render(<OrderedList editor={result.current!} />);

    expect(screen.getByRole("button")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button"));

    expect(container).toMatchSnapshot();
  });

  it.skip("render a CodeBlock button", () => {
    const { container } = render(<CodeBlock editor={result.current!} />);

    expect(screen.getByRole("button")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button"));

    expect(container).toMatchSnapshot();
  });

  it.skip("render a Blockquote button", () => {
    const { container } = render(<Blockquote editor={result.current!} />);

    expect(screen.getByRole("button")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button"));

    expect(container).toMatchSnapshot();
  });

  it.skip("render a Blockquote button", () => {
    const { container } = render(<Blockquote editor={result.current!} />);

    expect(screen.getByRole("button")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button"));

    expect(container).toMatchSnapshot();
  });
});
