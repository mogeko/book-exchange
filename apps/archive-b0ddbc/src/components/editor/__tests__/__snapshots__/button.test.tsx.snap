// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons render a Blockquote button 1`] = `
<div>
  <button
    class="btn btn-xs"
  >
    <svg
      class="w-3 h-3"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M464 256h-80v-64c0-35.3 28.7-64 64-64h8c13.3 0 24-10.7 24-24V56c0-13.3-10.7-24-24-24h-8c-88.4 0-160 71.6-160 160v240c0 26.5 21.5 48 48 48h128c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48zm-288 0H96v-64c0-35.3 28.7-64 64-64h8c13.3 0 24-10.7 24-24V56c0-13.3-10.7-24-24-24h-8C71.6 32 0 103.6 0 192v240c0 26.5 21.5 48 48 48h128c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48z"
      />
    </svg>
  </button>
</div>
`;

exports[`Buttons render a Blockquote button 2`] = `
<div>
  <button
    class="btn-active btn btn-xs"
  >
    <svg
      class="w-3 h-3"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M464 256h-80v-64c0-35.3 28.7-64 64-64h8c13.3 0 24-10.7 24-24V56c0-13.3-10.7-24-24-24h-8c-88.4 0-160 71.6-160 160v240c0 26.5 21.5 48 48 48h128c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48zm-288 0H96v-64c0-35.3 28.7-64 64-64h8c13.3 0 24-10.7 24-24V56c0-13.3-10.7-24-24-24h-8C71.6 32 0 103.6 0 192v240c0 26.5 21.5 48 48 48h128c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48z"
      />
    </svg>
  </button>
</div>
`;

exports[`Buttons render a BlueMark button 1`] = `
<div>
  <button
    class="btn btn-xs"
  >
    <svg
      class="w-3 h-3 text-[#74C0FC]"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
      />
    </svg>
  </button>
</div>
`;

exports[`Buttons render a Bold button 1`] = `
<div>
  <button
    class="btn btn-xs"
  >
    <svg
      class="w-3 h-3"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 384 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M333.49 238a122 122 0 0 0 27-65.21C367.87 96.49 308 32 233.42 32H34a16 16 0 0 0-16 16v48a16 16 0 0 0 16 16h31.87v288H34a16 16 0 0 0-16 16v48a16 16 0 0 0 16 16h209.32c70.8 0 134.14-51.75 141-122.4 4.74-48.45-16.39-92.06-50.83-119.6zM145.66 112h87.76a48 48 0 0 1 0 96h-87.76zm87.76 288h-87.76V288h87.76a56 56 0 0 1 0 112z"
      />
    </svg>
  </button>
</div>
`;

exports[`Buttons render a BulletList button 1`] = `
<div>
  <button
    class="btn btn-xs"
  >
    <svg
      class="w-3 h-3"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M48 48a48 48 0 1 0 48 48 48 48 0 0 0-48-48zm0 160a48 48 0 1 0 48 48 48 48 0 0 0-48-48zm0 160a48 48 0 1 0 48 48 48 48 0 0 0-48-48zm448 16H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
      />
    </svg>
  </button>
</div>
`;

exports[`Buttons render a CodeBlock button 1`] = `
<div>
  <button
    class="btn btn-xs"
  >
    <svg
      class="w-3 h-3"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 640 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M278.9 511.5l-61-17.7c-6.4-1.8-10-8.5-8.2-14.9L346.2 8.7c1.8-6.4 8.5-10 14.9-8.2l61 17.7c6.4 1.8 10 8.5 8.2 14.9L293.8 503.3c-1.9 6.4-8.5 10.1-14.9 8.2zm-114-112.2l43.5-46.4c4.6-4.9 4.3-12.7-.8-17.2L117 256l90.6-79.7c5.1-4.5 5.5-12.3.8-17.2l-43.5-46.4c-4.5-4.8-12.1-5.1-17-.5L3.8 247.2c-5.1 4.7-5.1 12.8 0 17.5l144.1 135.1c4.9 4.6 12.5 4.4 17-.5zm327.2.6l144.1-135.1c5.1-4.7 5.1-12.8 0-17.5L492.1 112.1c-4.8-4.5-12.4-4.3-17 .5L431.6 159c-4.6 4.9-4.3 12.7.8 17.2L523 256l-90.6 79.7c-5.1 4.5-5.5 12.3-.8 17.2l43.5 46.4c4.5 4.9 12.1 5.1 17 .6z"
      />
    </svg>
  </button>
</div>
`;

exports[`Buttons render a GreenMark button 1`] = `
<div>
  <button
    class="btn btn-xs"
  >
    <svg
      class="w-3 h-3 text-[#8CE99A]"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
      />
    </svg>
  </button>
</div>
`;

exports[`Buttons render a H1 button 1`] = `
<div>
  <button
    class="btn btn-xs"
  >
    H1
  </button>
</div>
`;

exports[`Buttons render a H2 button 1`] = `
<div>
  <button
    class="btn btn-xs"
  >
    H2
  </button>
</div>
`;

exports[`Buttons render a H3 button 1`] = `
<div>
  <button
    class="btn btn-xs"
  >
    H3
  </button>
</div>
`;

exports[`Buttons render a Italic button 1`] = `
<div>
  <button
    class="btn btn-xs"
  >
    <svg
      class="w-3 h-3"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 320 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M320 48v32a16 16 0 0 1-16 16h-62.76l-80 320H208a16 16 0 0 1 16 16v32a16 16 0 0 1-16 16H16a16 16 0 0 1-16-16v-32a16 16 0 0 1 16-16h62.76l80-320H112a16 16 0 0 1-16-16V48a16 16 0 0 1 16-16h192a16 16 0 0 1 16 16z"
      />
    </svg>
  </button>
</div>
`;

exports[`Buttons render a OrderedList button 1`] = `
<div>
  <button
    class="btn btn-xs"
  >
    <svg
      class="w-3 h-3"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M61.77 401l17.5-20.15a19.92 19.92 0 0 0 5.07-14.19v-3.31C84.34 356 80.5 352 73 352H16a8 8 0 0 0-8 8v16a8 8 0 0 0 8 8h22.83a157.41 157.41 0 0 0-11 12.31l-5.61 7c-4 5.07-5.25 10.13-2.8 14.88l1.05 1.93c3 5.76 6.29 7.88 12.25 7.88h4.73c10.33 0 15.94 2.44 15.94 9.09 0 4.72-4.2 8.22-14.36 8.22a41.54 41.54 0 0 1-15.47-3.12c-6.49-3.88-11.74-3.5-15.6 3.12l-5.59 9.31c-3.72 6.13-3.19 11.72 2.63 15.94 7.71 4.69 20.38 9.44 37 9.44 34.16 0 48.5-22.75 48.5-44.12-.03-14.38-9.12-29.76-28.73-34.88zM496 224H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zM16 160h64a8 8 0 0 0 8-8v-16a8 8 0 0 0-8-8H64V40a8 8 0 0 0-8-8H32a8 8 0 0 0-7.14 4.42l-8 16A8 8 0 0 0 24 64h8v64H16a8 8 0 0 0-8 8v16a8 8 0 0 0 8 8zm-3.91 160H80a8 8 0 0 0 8-8v-16a8 8 0 0 0-8-8H41.32c3.29-10.29 48.34-18.68 48.34-56.44 0-29.06-25-39.56-44.47-39.56-21.36 0-33.8 10-40.46 18.75-4.37 5.59-3 10.84 2.8 15.37l8.58 6.88c5.61 4.56 11 2.47 16.12-2.44a13.44 13.44 0 0 1 9.46-3.84c3.33 0 9.28 1.56 9.28 8.75C51 248.19 0 257.31 0 304.59v4C0 316 5.08 320 12.09 320z"
      />
    </svg>
  </button>
</div>
`;

exports[`Buttons render a RedMark button 1`] = `
<div>
  <button
    class="btn btn-xs"
  >
    <svg
      class="w-3 h-3 text-[#FFA8A8]"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
      />
    </svg>
  </button>
</div>
`;

exports[`Buttons render a Strike button 1`] = `
<div>
  <button
    class="btn btn-xs"
  >
    <svg
      class="w-3 h-3"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M496 224H293.9l-87.17-26.83A43.55 43.55 0 0 1 219.55 112h66.79A49.89 49.89 0 0 1 331 139.58a16 16 0 0 0 21.46 7.15l42.94-21.47a16 16 0 0 0 7.16-21.46l-.53-1A128 128 0 0 0 287.51 32h-68a123.68 123.68 0 0 0-123 135.64c2 20.89 10.1 39.83 21.78 56.36H16a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h480a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm-180.24 96A43 43 0 0 1 336 356.45 43.59 43.59 0 0 1 292.45 400h-66.79A49.89 49.89 0 0 1 181 372.42a16 16 0 0 0-21.46-7.15l-42.94 21.47a16 16 0 0 0-7.16 21.46l.53 1A128 128 0 0 0 224.49 480h68a123.68 123.68 0 0 0 123-135.64 114.25 114.25 0 0 0-5.34-24.36z"
      />
    </svg>
  </button>
</div>
`;

exports[`Buttons render a YellowMark button 1`] = `
<div>
  <button
    class="btn btn-xs"
  >
    <svg
      class="w-3 h-3 text-[#FFC078]"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
      />
    </svg>
  </button>
</div>
`;
